### PR TITLE
US006 Adding Responsibilities Per Role - API

### PR DESCRIPTION
**Pull request by CharlesM988**
Adding in functionality to push Responsibilities Per Role to view-job-spec.html

- Created ResponsibilityService
- Created Responsibility
- Created ResponsibilityDao
- Created ResponsibilityController
- Created RoleResponsibilityDoesNotExistException
- FailedToGetRoleResponsibilityException
- Modified DropwizardApplication

Screenshot 2023-12-20 at 10 51 50

Screenshot 2023-12-20 at 10 52 21

Screenshot 2023-12-20 at 10 49 53

### CODE REVIEW

**Review by amchugh689 — APPROVED** [review]
tested & working !

File: src/main/java/org/kainos/ea/cli/Responsibility.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+public class Responsibility {
+    private String responsibilityName;
+
+    public Responsibility() {
+        this.responsibilityName = responsibilityName;
+    }
+
+    public String getResponsibilityName() {
+
```
Comment (jkelso662) [inline src/main/java/org/kainos/ea/cli/Responsibility.java]: Does your linted not complain about these white spaces?

**Review by jkelso662 — APPROVED** [review]
Have a look at my wee comment but other than that - LGTM!

File: src/main/java/org/kainos/ea/cli/Responsibility.java (hunk 1 above)
Comment (CharlesM988) [reply src/main/java/org/kainos/ea/cli/Responsibility.java]: @jkelso662 No it doesn't seem to mind about those whitespaces. I've removed them and I'll be more mindful in the future.

**Pull request merged**